packages/luna/src/hooks/auth.ts

```diff
@@ -0,0 +1,26 @@
+import { Void } from "@litespace/types";
+import { useMutation } from "@tanstack/react-query";
+import { useCallback } from "react";
+import { atlas } from "@/lib/atlas";
+
+type OnSuccess = Void;
+type OnError = (error: Error) => void;
+
+export function useVerifyEmail({
+  onSuccess,
+  onError,
+}: {
+  onSuccess: OnSuccess;
+  onError: OnError;
+}) {
+  const verifyEmail = useCallback((token: string) => {
+    return atlas.auth.verifyEmail(token);
+  }, []);
```

Comment:
You are missing a dependency in your `useCallback`. You can use `eslint` to help you not to forget to add you dependencies.